funcx_sdk/funcx/sdk/executor.py

```diff
@@ -226,17 +334,97 @@ def _submit_tasks(self, messages):
                 raise
             else:
                 for i, msg in enumerate(messages):
-                    self._function_future_map[batch_tasks[i]] = self._tasks.pop(
+                    task_uuid: str = batch_tasks[i]
+                    self._function_future_map[task_uuid] = self._tasks.pop(
                         msg["task_id"]
                     )
-                    self.poller_thread.atomic_controller.increment()
+                assert self.poller_thread is not None  # Satisfy linter
+                self.poller_thread.atomic_controller.increment(val=len(messages))
+
+    def reload_tasks(self) -> t.Iterable[Future]:
+        """
+        Load the set of tasks associated with this Executor's Task Group (FuncXClient)
+        from the server and return a set of futures, one for each task.  This is
+        nominally intended to "reattach" to a previously initiated session, based on
+        the Task Group ID.  An example use might be:
+
+            import sys
+            import typing as T
+            from funcx import FuncXClient
+            from funcx.sdk.executor import FuncXExecutor
+
+            fxc_kwargs = {}
+            if len(sys.argv) > 1:
+                fxc_kwargs["task_group_id"] = sys.argv[1]
+                task_group_id = sys.argv[1]
+
+            fxclient = FuncXClient(**fxc_kwargs)
+            fxexec = FuncXExecutor(fxclient)
+
+            # Save the task_group_id somewhere.  Perhaps in a file, or less
+            # robustly "as mere text" on your console:
+            print("If this script dies, rehydrate futures with this "
+                 f"Task Group ID: {fxexec.task_group_id}")
+
+            futures: T.Iterable[Future] = []
+            if fxc_kwargs:
+                print(f"Reloading tasks associated with Task Group ID: {sys.argv[1]}")
+                futures = fxexec.reload_tasks()
+            else:
+                def example_funcx_kernel():
```

Comment:
I get a complaint with `example_funcx_kernel` being serialized since it is indented within the if-else, that is a separate problem but easiest thing to do is move the function definition up above the if-else

Reply:
Okay; I'll try out this example I wrote specifically, but the point wasn't for a complete ready-to-run snippet of code, but to show the general thought-process.